implementations/rust/ockam/ockam_transport_tcp/tests/send_receive.rs

```diff
@@ -1,33 +1,27 @@
-use core::iter;
-
+use ockam_core::compat::rand::{self, Rng};
 use ockam_core::{route, Address, Result, Routed, Worker};
 use ockam_node::Context;
-use rand::Rng;
 
 use ockam_transport_tcp::{TcpTransport, TCP};
 
 #[ockam_macros::test]
 async fn send_receive(ctx: &mut Context) -> Result<()> {
-    let rand_port = rand::thread_rng().gen_range(10000, 65535);
-    let bind_address = format!("127.0.0.1:{}", rand_port);
-    let bind_address = bind_address.as_str();
+    let listener_address;
 
     let _listener = {
         let transport = TcpTransport::create(ctx).await?;
-        transport.listen(bind_address).await?;
+        listener_address = transport.listen("127.0.0.1:0").await?;
         ctx.start_worker("echoer", Echoer).await?;
     };
```

Comment:
How about removing the `let _listener` block and initialising `listener_address` directly? E.g.

```rust
let transport = TcpTransport::create(ctx).await?;
let listener_address = transport.listen("127.0.0.1:0").await?;
ctx.start_worker("echoer", Echoer).await?;
```